pkg/api/service.go

```diff
@@ -225,12 +238,89 @@ func (s *Service) PublishEnvelope(
 	return &message_api.PublishEnvelopeResponse{OriginatorEnvelope: originatorEnv}, nil
 }
 
+func (s *Service) maybePublishToBlockchain(
+	ctx context.Context,
+	clientEnv *message_api.ClientEnvelope,
+) (didPublish bool, err error) {
+	identityUpdate := clientEnv.GetIdentityUpdate()
+	if identityUpdate != nil {
+		if err = s.publishIdentityUpdate(ctx, identityUpdate); err != nil {
+			s.log.Error("could not publish identity update", zap.Error(err))
+			return false, status.Errorf(
+				codes.Internal,
+				"could not publish identity update: %v",
+				err,
+			)
+		}
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (s *Service) publishIdentityUpdate(
+	ctx context.Context,
+	identityUpdate *associations.IdentityUpdate,
+) error {
+	identityUpdateBytes, err := proto.Marshal(identityUpdate)
+	if err != nil {
+		return err
+	}
+	inboxId, err := utils.ParseInboxId(identityUpdate.InboxId)
+	if err != nil {
+		return err
+	}
+	return s.messagePublisher.PublishIdentityUpdate(
+		ctx,
+		inboxId,
+		identityUpdateBytes,
+	)
+}
+
+func (s *Service) GetInboxIds(
+	ctx context.Context,
+	req *message_api.GetInboxIdsRequest,
+) (*message_api.GetInboxIdsResponse, error) {
```

Comment:
How would you feel about us replacing this with a `batchQuery` endpoint eventually? I'm also curious if for the demo the client could have done the batching internally with multiple requests

Reply:
It would have to be its own`batchQuery` endpoint, since the address log is separate from the gateway envelopes table.